src/ReactSVG.tsx

```diff
@@ -70,7 +70,12 @@ export class ReactSVG extends React.Component<Props, State> {
       const wrapper = this.props.wrapper!
       /* eslint-enable @typescript-eslint/no-non-null-assertion */
 
-      const nonReactElement = document.createElement(wrapper)
+      // createElement seems to work for svg, but createElementNS with an SVG namespace may be called for in some cases?
```

Comment:
Are you talking about this note from https://developer.mozilla.org/en-US/docs/Web/SVG/Element/svg?

> Note: The xmlns attribute is only required on the outermost svg element of SVG documents. It is unnecessary for inner svg elements or inside HTML documents.

Assuming "unnecessary" means it won't throw an error, I wonder if it's easier to just always add the namespace which will cover all scenarios?

So:

```ts
const nonReactElement =
        wrapper === 'svg'
          ? document.createElementNS('http://www.w3.org/2000/svg', wrapper)
          : document.createElement(wrapper)
```

Which reminds me, that React-managed outer wrapper will need something similar done too I imagine 🤔 Will have a play around.

Reply:
Done in 1d072a8.

Reply:
I wasn't talking about that comment, no. I'm having trouble finding **authoritative** documentation on when/why to use createElement vs. createElementNS, but I guess https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement and https://developer.mozilla.org/en-US/docs/Web/API/Document/createElementNS may be as close as I can get.

However, here's a brief discussion of createElement vs createElementNS for svgs: https://webhint.io/docs/user-guide/hints/hint-create-element-svg/

I can say that whether you use createElementNS or createElement does not impact whether the xmlns attribute appears. To add that, you can use code like:

```
  if (wrapper === 'svg') {
    nonReactElement.setAttribute("xmlns", "http://www.w3.org/2000/svg")
    nonReactElement.setAttribute("version", "1.1")
  }
```

(I included the version as well, since at some point I decided to do that as a matter of course whenever I create an SVG dynamically. Note that I also believe but cannot reconstruct my documentation trail that these should be set with `setAttribute` and not with `setAttributeNS`.)

Based on the fact that SVGs were designed to be composable, I would think that including these attributes on nested SVGs is not a problem.

And.. yeah. Wherever the SVGs are created, I'd expect this would be a potential problem. (If they're parsed in, however, the parser should handle the namespace itself, which IIRC you already know better than me and are doing in your SVG injector!)

Reply:
Good points, looks slightly related to the other question I just commented on, but my take on these:

- Makes sense to use `createElementNS` for SVGs.
- You're right about the xmlns attribute, and yea `@tanem/svg-injector` sets this using `setAttribute` for the injected SVG, so probably makes sense to set it in the same way on the wrapper. Useful text [here](https://developer.mozilla.org/en-US/docs/Web/SVG/Namespaces_Crash_Course#scripting_in_namespaced_xml) from MDN, IIRC using `null` as the first param to `setAttributeNS` is equivalent to just using `setAttribute`.